client/extensions/woocommerce/app/settings/shipping/shipping-zone/shipping-zone-location-dialog-settings.js

```diff
@@ -0,0 +1,185 @@
+/* eslint wpcalypso/i18n-ellipsis: 0 */
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import FilteredList from 'woocommerce/components/filtered-list';
+import FormCheckbox from 'components/forms/form-checkbox';
+import FormFieldSet from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormRadio from 'components/forms/form-radio';
+import FormTextArea from 'components/forms/form-textarea';
+import { bindActionCreatorsWithSiteId } from 'woocommerce/lib/redux-utils';
+import { decodeEntities } from 'lib/formatting';
+import {
+	canLocationsBeFiltered,
+	canLocationsBeFilteredByState,
+	areLocationsFilteredByPostcode,
+	areLocationsFilteredByState,
+	getCurrentlyEditingShippingZoneStates,
+} from 'woocommerce/state/ui/shipping/zones/locations/selectors';
+import {
+	filterByWholeCountry,
+	filterByState,
+	filterByPostcode,
+	toggleStateSelected,
+	editPostcode,
+} from 'woocommerce/state/ui/shipping/zones/locations/actions';
+
+const ShippingZoneLocationDialogSettings = ( {
+		canFilter,
+		filteredByPostcode,
+		canFilterByState,
+		filteredByState,
+		states,
+		translate,
+		actions,
+	} ) => {
+	const onWholeCountrySelect = ( event ) => {
+		event.stopPropagation();
+		actions.filterByWholeCountry();
+	};
+	const onStateSelect = ( event ) => {
+		event.stopPropagation();
+		actions.filterByState();
+	};
+	const onPostcodeSelect = ( event ) => {
+		event.stopPropagation();
+		actions.filterByPostcode();
+	};
+
+	const renderRadios = () => {
+		if ( ! canFilter ) {
+			return (
+				<div>
+					<FormRadio disabled={ true } checked={ true } readOnly={ true } />
+					{ translate( 'Include entire countries in the zone' ) }
+				</div>
+			);
+		}
+
+		const radios = [
+			<div key={ 0 } onClick={ onWholeCountrySelect }>
+				<FormRadio
+					onChange={ onWholeCountrySelect }
+					checked={ ! filteredByPostcode && ! filteredByState } />
+				{ translate( 'Include entire country in the zone' ) }
```

Comment:
If `getCurrentSelectedCountryZoneOwner` returns "true-ish", then you can't filter by "whole country". It's an edge case you can trigger by:
* Create a zone with a country, like `US`. Save it.
* Edit another zone. You will be able to select `US` only if there are no other countries or continents selected. And then, you will be able to only filter by states or postcode, because the "full country" already has an owner.